examples/dlint/main.rs

```diff
@@ -35,7 +36,41 @@ fn create_cli_app<'a, 'b>() -> App<'a, 'b> {
     )
 }
 
-fn format_diagnostic(diagnostic: &LintDiagnostic, source: &str) -> String {
+fn get_slice_source_and_range<'a>(
+  source: &'a str,
+  range: &Range,
+) -> (&'a str, (usize, usize)) {
+  let line_start_indexes = std::iter::once(0)
+    .chain(source.match_indices('\n').map(|l| l.0 + 1))
```

Comment:
Just a heads up, this is incorrect because ECMAScript defines `\n`, `\r`, `\u{2028}`, and `\u{2029}` all as valid linebreak characters (although we havent fixed it in rslint yet either 😄)

Reply:
@RDambrosio016 - thanks for pointing this out. I'll open an issue about it. All in all `annotate-snippets` is pretty cool, although I found the need to manually take slices of source code and adjusting ranges a bit cumbersome... Otherwise it results in output as seen in https://github.com/denoland/deno_lint/pull/391#issue-502664928

Reply:
Thats kind of odd, it sounds like the ranges are a bit broken, no?

Reply:
It does sound like it - I checked `annotate-snippets` issues but couldn't find anything about it, I guess that's just how the crate works ¯\_(ツ)_/¯ anyway, taking a slice and adjusting range is no big deal so that's fine.